projects/fw102_hw/src/main.c

```diff
@@ -0,0 +1,49 @@
+#include <stdint.h>  // for integer types
+#include <stdlib.h>  // for random numbers
+
+#include "interrupt.h"   // interrupts are required for soft timers
+#include "log.h"         // for printing
+#include "soft_timer.h"  // for soft timers
+#include "wait.h"        // for wait function
+
+typedef struct {
```

Comment:
By C convention the typename is also included in the typedef line, e.g. `typedef struct Counters {`